src/tempdir.js

```diff
@@ -12,10 +20,8 @@ function writeableDir(dir) {
   if (!fs.statSync(dir).isDirectory())
     return false;
 
-  var testFile = dir+'/'+common.randomFileName();
   try {
-    fs.writeFileSync(testFile, ' ');
-    common.unlinkSync(testFile);
+    getMktemp()('-u', path.resolve(dir, 'tmp.shelljs.XXXXXXXXXX'));
```

Comment:
Also here, this should be updated to `({ unsafe: true }, ...)` instead of `('-u', ...)`